### DISCUSSION TITLE
CircleCI executor dependecy detection

### DISCUSSION THREAD

**Discussion by tlbraams** [renovatebot/renovate#30558] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

In a recent ruby update PR, I discovered that the CircleCI convenience image (cimg/ruby) was not updated or even recognized. As previous PRs did include updates for this image, I investigated how/when/why this broke. It turns out that after a refactor (#30142), the CircleCI manager no longer detects any docker images configured in executors.

I've created a reproduction that uses the example configuration provided in the CircleCI documentation: https://github.com/tlbraams/renovate-circleci-executors
In the readme I included logs from the last renovate version that detects the dependency (37.429.0) and the version that no longer detects the dependency (37.431.0) as well as a recently released version (38.17.0)

**Comment by secustor** [comment, marked as answer]
Thanks for the good reproduction, I have created a fix for it: https://github.com/renovatebot/renovate/pull/30562
